File: src/main/java/gargoyle/xenox/util/config/MemoryPreferencesFactory.java
package gargoyle.xenox.util.config;

import java.util.prefs.Preferences;
import java.util.prefs.PreferencesFactory;

class MemoryPreferencesFactory implements PreferencesFactory {
    private final MemoryPreferences systemRoot = new MemoryPreferences(null, "");
    private final MemoryPreferences userRoot = new MemoryPreferences(null, "");

    @Override
    public Preferences systemRoot() {
        return systemRoot;
    }

    @Override
    public Preferences userRoot() {
        return userRoot;
    }
}

File: src/main/kotlin/gargoyle/xenox/util/config/MemoryPreferencesFactory.kt
package gargoyle.xenox.util.config

import java.util.prefs.Preferences
import java.util.prefs.PreferencesFactory

internal class MemoryPreferencesFactory : PreferencesFactory {
    private val systemRoot = MemoryPreferences(null, "")
    private val userRoot = MemoryPreferences(null, "")
    override fun systemRoot(): Preferences {
        return systemRoot
    }

    override fun userRoot(): Preferences {
        return userRoot
    }
}
